osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -109,7 +109,15 @@ public string ModsJson
         public string UserString
         {
             get => User?.Username;
-            set => User = new User { Username = value };
+            set => User = new User { Username = value, Id = UserID};
+        }
+
+        [JsonIgnore]
+        [Column("UserID")]
+        public long UserID
```

Comment:
do these have to be public for serialization to database?

Reply:
I just gave it a test, it does have to be public for serialization to database